### DISCUSSION TITLE
V14 detect steam switching and clips folder usage

### DISCUSSION THREAD

**Discussion by ediguidedog** [blakeblackshear/frigate#12671] [General]
I've been playing around with V14 on a spare box I have so I'm relying on ov for my object detection and because of that I thought I should look into using a sub streams for detection purposes to try and reduce the overall cpu load. My sub streams are D1 and have a 4:3 aspect ratio compared to the FHD primary stream. This has the disadvantage that things look a bit odd in the console (fixable with config I'm sure) but the advantage I can easily see when frigate switches from sub stream to main stream.

What I am noticing is that it seems to switch when movement is detected rather than when an object is actually recognised. Is this intended behaviour?  Can this be configured to switch when an object of interest is actually detected (regardless of object location in frame) rather than simply movement as I find it really distracting waving branches and bushes constantly triggering when it's windy?

The other question I have is around the clips folder. This seems to be hogging disk space at the expense of the recordings folder. Am I right in thinking this is some sort of 'not much happening here' sort of store which does however allow me to scroll between events and have a every second or two view what's been happening (or not) when there are no events.  So what's happening on my test box is I have quite a small SSD (128GB inc os) and the clips folder is consuming half of the SSD (50 odd GB) leaving only 13GB free for the important recordings.  Is there a way to limit the clips folder to only a few GB so the more useful recordings can have more space.  Is it also possible to show the clips folder usage on the storage pages as with it being absent the page currently under reports Frigates actual disc usage?

**Comment by hawkeye217** [comment]
From the release notes:

> Your camera images update once per minute when no detectable activity is occurring to conserve bandwidth and resources. As soon as any motion is detected, cameras seamlessly switch to a full-resolution live stream.

So yes, what you are seeing is intended behavior. Cameras on the Live dashboard switch from stills to videos on motion. If your aspect ratios are different between your two streams, what you're experiencing is also expected. 

Regarding the clips folder, it'd be helpful to see your Frigate config file.

**Reply by ediguidedog** [reply]
I get the update every minute but that doesn't need to switch to the main HD feed. What I don't understand is why switch to the main feed simply for motion? Surely it would be better to switch if there was actually something useful to see (so a detected object) rather that something like a bush blowing in the wind.

I'm assuming the aspect ratio can be fixed with little CPU overhead but I have to admit I haven't actually looked yet as its useful to see at a glance when it's moving between the streams.

**Reply by NickM-27** [reply]
it's funny, you say that but then other users ask why not just show live all the time because they never want to miss anything. Switching on motion is a good middle ground that ensures fast moving objects can be seen as soon as possible. Otherwise, there is a chance that the object takes more frames than normal to be detected as an object and by that time it is almost out of the frame. The view does specifically highlight the camera with a red outline only when an active object is detected

**Reply by hawkeye217** [reply]
It's also worth mentioning that Birdseye is still available in 0.14 and can cover your objects-only use case.

```
birdseye:
  mode: objects
```

**Reply by ediguidedog** [reply]
> it's funny, you say that but then other users ask why not just show live all the time because they never want to miss anything. Switching on motion is a good middle ground that ensures fast moving objects can be seen as soon as possible. Otherwise, there is a chance that the object takes more frames than normal to be detected as an object and by that time it is almost out of the frame. The view does specifically highlight the camera with a red outline only when an active object is detected

You are right in so much as when I sort the aspect ratio out of the detect stream in the UI it will be a lot less annoying so I can see both sides here. 

> It's also worth mentioning that Birdseye is still available in 0.14 and can cover your objects-only use case.
> 
> ```
> birdseye:
>   mode: objects
> ```

Thanks. That's good to know as it was a bit of a pain before on windy days

**Comment by a-lunatic** [comment]
I moved the clips and export folder to a other drive as it started taking up a lot of space and to allow for more recording space so if possible it would be nice to see how much space it takes up on the Storage page.

**Reply by NickM-27** [reply]
it's not really possible because the sizes are not stored in the DB so calculating the size would require spinning up the disk

**Reply by ediguidedog** [reply]
So what actually dictates how large this folder can grow as with finite storage it just seems to keep growing at the expense of the space available for my more useful recordings?

**Reply by NickM-27** [reply]
The only things in the /clips folder are:

1. snapshots (retained per your snapshot config)
2. review item thumbnails (retained per recordings config)
3. preview videos (retained per recordings config)

This is how the breakdown is for me with keeping snapshots and continuous recordings (all) for 14 days on 8 cameras

recordings take up 4.24 TB

1. snapshots 4 GB
2. review thumbs 15 MB
3. preview videos 8 GB

**Reply by ediguidedog** [reply]
> The only things in the /clips folder are:
> This is how the breakdown is for me with keeping snapshots and continuous recordings (all) for 14 days on 8 cameras
> 
> recordings take up 4.24 TB
> 
> 1. snapshots 4 GB
> 2. review thumbs 15 MB
> 3. preview videos 8 GB

So I have 
344MB in Previews 
6.2MB In Review
And a whopping 44GB of JPG and PNG files in the clips folder. 

I think I can see my problem here in that in my current live setup I have snapshots (for people at least) set to 28 days and this gives a about 20GB of snapshots which go back much further then my event recordings but in V14 the same number of days doubles this storage requirements as it appears the PNG files are about twice the size now from what the were in V12. Is there a reason for this?  Is there any point in V14 actually keeping snapshots at all as I cant see a way via the GUI to see them?

**Reply by hawkeye217** [reply]
If you're not using Frigate+ and don't have a need for snapshots elsewhere, you can certainly disable them.

**Reply by ediguidedog** [reply]
> If you're not using Frigate+ and don't have a need for snapshots elsewhere, you can certainly disable them.

Thanks. I've disabled them but I did have to drop into the file system and manually delete all the snapshots that were already there as I guess the maintenance routine that deleted snapshots over x days doesn't actually run if the snapshots are disabled
